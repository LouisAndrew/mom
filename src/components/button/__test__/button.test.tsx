import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Icon } from '@iconify/react';
import adobeIcon from '@iconify/icons-uil/adobe';

// import Button from '..';
import Button from '..';

describe('Button element', () => {
    const mockHandleClick = jest.fn(() => {});
    const mockInnerText: string = 'Click me!';

    afterEach(cleanup);
    const el: React.ReactElement = (
        <Button handleClick={mockHandleClick}>{mockInnerText}</Button>
    );
    const elWithIcon: React.ReactElement = (
        <Button handleClick={mockHandleClick}>
            <Icon data-testid="icon" icon={adobeIcon} />
            {mockInnerText}
        </Button>
    );
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(el, div);
    });

    it('should render and react to user changes correctly', () => {
        const { getByText } = render(el);

        const buttonEl: HTMLElement = getByText(mockInnerText);
        fireEvent.click(buttonEl);
        expect(mockHandleClick).toBeCalled();
    });

    it('should render with icon correctly', () => {
        const { queryByTestId } = render(elWithIcon);
        const iconEl: HTMLElement | null = queryByTestId('icon');

        if (iconEl) {
            // if query is successful = test successful!
            expect(iconEl).toBeInTheDocument();
        } else fail();
    });

    it('matches snapshot', () => {
        const tree = renderer.create(el).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
