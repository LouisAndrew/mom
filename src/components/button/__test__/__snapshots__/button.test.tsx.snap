// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button element matches snapshot 1`] = `
<button
  className="sc-AxjAm eBSGmW"
  fontFamily="body"
  fontSize={
    Array [
      1,
      1,
      2,
      2,
    ]
  }
  fontWeight="bold"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [
          Class {
            "_dispatchInstances": null,
            "_dispatchListeners": null,
            "_targetInst": null,
            "altKey": null,
            "bubbles": null,
            "button": null,
            "buttons": null,
            "cancelable": null,
            "clientX": null,
            "clientY": null,
            "ctrlKey": null,
            "currentTarget": [Function],
            "defaultPrevented": null,
            "detail": null,
            "dispatchConfig": null,
            "eventPhase": null,
            "getModifierState": [Function],
            "isDefaultPrevented": [Function],
            "isPropagationStopped": [Function],
            "isTrusted": null,
            "metaKey": null,
            "movementX": [Function],
            "movementY": [Function],
            "nativeEvent": null,
            "pageX": null,
            "pageY": null,
            "relatedTarget": [Function],
            "screenX": null,
            "screenY": null,
            "shiftKey": null,
            "target": null,
            "timeStamp": [Function],
            "type": null,
            "view": null,
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  Click me!
</button>
`;
